File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/data/local/LocalRepository.java
package org.mifos.mobilewallet.mifospay.data.local;

import javax.inject.Inject;
import javax.inject.Singleton;

import org.mifos.mobilewallet.core.domain.model.Client;

/**
 * Created by naman on 17/6/17.
 */

@Singleton
public class LocalRepository {

    private final PreferencesHelper preferencesHelper;

    @Inject
    public LocalRepository(PreferencesHelper preferencesHelper) {
        this.preferencesHelper = preferencesHelper;
    }

    public Client getClientDetails() {
        Client details = new Client();
        details.setName(preferencesHelper.getFullName());
        details.setClientId(preferencesHelper.getClientId());

        return details;
    }

    public void saveClientData(Client client) {
        preferencesHelper.saveFullName(client.getName());
        preferencesHelper.setClientId(client.getClientId());
    }


}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/data/local/LocalRepository.kt
package org.mifos.mobilewallet.mifospay.data.local

import org.mifos.mobilewallet.core.data.fineract.local.PreferencesHelper
import com.mifos.mobilewallet.model.domain.client.Client
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by naman on 17/6/17.
 */
@Singleton
class LocalRepository @Inject constructor(val preferencesHelper: PreferencesHelper) {

    val clientDetails: com.mifos.mobilewallet.model.domain.client.Client
        get() {
            val details = com.mifos.mobilewallet.model.domain.client.Client()
            details.name = preferencesHelper.fullName
            details.clientId = preferencesHelper.clientId
            details.externalId = preferencesHelper.clientVpa
            return details
        }

    fun saveClientData(client: com.mifos.mobilewallet.model.domain.client.Client) {
        preferencesHelper.saveFullName(client.name)
        preferencesHelper.clientId = client.clientId
        preferencesHelper.clientVpa = client.externalId
    }
}